src/bin/scheduler/setup/general.rs

```diff
@@ -183,6 +187,61 @@ fn setup_rcc_working_directories(
     )
 }
 
+fn setup_environment_building_directories(
```

Comment:
Why is there so much extra code here? Is this change adding functionality?

Reply:
Not really, but I found it hard to consolidate this with the already existing functionality. I would suggest that we first make the necessary changes and then check if and how we can reduce code duplication. Ok?

Reply:
Yes, that does not bother me. It's not really about the code duplication, but I think we might need better interfaces for adding functionality.